examples/cosmos/ics20/ics20.qnt

```diff
@@ -0,0 +1,129 @@
+// -*- mode: Bluespec; -*-
+
+/**
+ * A specification ICS20 fungible token transfer protocol.
+ *
+ * For now, this only models the denomination trace changes between transfers.
+ *
+ * Gabriela Moreira, Informal Systems, 2023
+ */
+module ics20 {
+  type Uint = int
+
+  type State = {
+    trace: List[str],
+    denom: str,
+  }
```

Comment:
Is there a reason to carry around `denom` at all?
It's only the base denomination, and we currently set it once in `init` and then just propagate it.

Reply:
> Is there a reason to carry around denom at all?

The combination of `denom` and `trace` in this type represents the `denom` string we see in IBC packets (i.e. `"transfer/channel/mydenom"`). Since we don't have string manipulation in quint, this is the data structure I came up with.

I'll rename those to look similar to the hashed version of this field, which has the following structure after decoding:
```go
// DenomTrace contains the base denomination for ICS20 fungible tokens and the source tracing
// information
message DenomTrace {
  // chain of port/channel identifiers used for tracing the source of the fungible token
  string path = 1;
  // base denomination of the relayed fungible token
  string base_denom = 2;
}
```

`path` will have to be a list to allow slicing and concatenation, but other than that, I think the names will work better than the ones I picked (from studying the old non-hashed implementation)